### ISSUE TITLE
Spaces in paths break adb commands on windows.

### ISSUE THREAD

**Issue by jtszalay** [ItsRiprod/DeskThing#21]
Windows in it's infinite wisdom named my user directory with a space:
```
C:\Users\James Szalay\AppData\Local\Programs\deskthing>
```

This often breaks apps that don't handle paths with spaces on windows.

**Comment by ItsRiprod** [comment]
Strange? That should be fixed in v0.7.0

**Comment by jtszalay** [comment]
```
[4:22:03 PM] ADB Error: , -s 8556R682QR1R push C:\Users\James Szalay\AppData\Roaming\deskthing\webapp/ /usr/share/qt-superbird-app/webapp, C:\Users\James Szalay\AppData\Local\Programs\deskthing\resources\win\adb.exe

[4:22:03 PM] HandlePushWebApp encountered the error ADB Error: , -s 8556R682QR1R push C:\Users\James Szalay\AppData\Roaming\deskthing\webapp/ /usr/share/qt-superbird-app/webapp, C:\Users\James Szalay\AppData\Local\Programs\deskthing\resources\win\adb.exe
```

That's the log I get. I'm able to manually run the commands adding quotes around my path and it works. 

![image](https://github.com/user-attachments/assets/e0f9140e-6887-480e-af26-463f0a985222)

**Comment by jtszalay** [comment]
```
diff --git a/DeskThingServer/src/main/handlers/adbHandler.ts b/DeskThingServer/src/main/handlers/adbHandler.ts
index acca2e0..b2935b4 100644
--- a/DeskThingServer/src/main/handlers/adbHandler.ts
+++ b/DeskThingServer/src/main/handlers/adbHandler.ts
@@ -11,9 +11,21 @@ const execPath = isDevelopment
 const adbExecutableName = process.platform === 'win32' ? 'adb.exe' : 'adb'
 const adbPath = path.join(execPath, adbExecutableName)
 
+const splitArgs = (str: string): string[] => {
+  const regex: RegExp = /(".*?"|[^"\s]+)(?=\s*|\s*$)/g;
+  const matches: string[] = [];
+  let match: RegExpExecArray | null;
+
+  while ((match = regex.exec(str)) !== null) {
+    matches.push(match[1].replace(/(^"|"$)/g, '')); // Remove surrounding quotes if any
+  }
+
+  return matches;
+};
+
 export const handleAdbCommands = (command: string, event?): Promise<string> => {
   return new Promise((resolve, reject) => {
-    execFile(adbPath, command.split(' '), { cwd: execPath }, (error, stdout, stderr) => {
+    execFile(adbPath, splitArgs(command), { cwd: execPath }, (error, stdout, stderr) => {
       if (error) {
         if (event) {
           event.sender.send('logging', {
diff --git a/DeskThingServer/src/main/handlers/deviceHandler.ts b/DeskThingServer/src/main/handlers/deviceHandler.ts
index f3a8eeb..9716549 100644
--- a/DeskThingServer/src/main/handlers/deviceHandler.ts
+++ b/DeskThingServer/src/main/handlers/deviceHandler.ts
@@ -64,7 +64,7 @@ export const HandlePushWebApp = async (
 
     reply('logging', { status: true, data: response || 'Pushing new app...', final: false })
     response = await handleAdbCommands(
-      `-s ${deviceId} push ${extractDir}/ /usr/share/qt-superbird-app/webapp`
+      `-s ${deviceId} push "${extractDir}/" /usr/share/qt-superbird-app/webapp`
     )
 
     reply('logging', { status: true, data: response || 'Restarting Chromium', final: false })

```

Wrapping with quotes and then splitting the string preserving the quotes for boundaries worked for me.

**Comment by ItsRiprod** [comment]
Interesting! I actually don't think I fully realized it wasn't the access to ADB that was the issue (hence using execFile rather than exec) but rather the file path being passed into the arguments! 

Good catch

Implemented in v0.8.0
